docs/src/pages/components/switches/CustomizedSwitches.tsx

```diff
@@ -1,129 +1,97 @@
-  createStyles({
-    root: {
-      width: 42,
-      height: 26,
-      padding: 0,
-      margin: theme.spacing(1),
+    '&.Mui-checked + .MuiSwitch-track': {
+      backgroundColor: purple[500],
     },
-    switchBase: {
-      padding: 1,
-      '&$checked': {
-        transform: 'translateX(16px)',
-        color: theme.palette.common.white,
-        '& + $track': {
-          backgroundColor: '#52d869',
-          opacity: 1,
-          border: 'none',
-        },
-      },
-      '&$focusVisible $thumb': {
-        color: '#52d869',
-        border: '6px solid #fff',
+  },
+}));
+
+const IOSSwitch = styled((props: SwitchProps) => (
+  <Switch
+    // focusVisibleClassName={classes.focusVisible}
+    disableRipple
+    {...props}
+  />
+))(({ theme }) => ({
+  width: 42,
+  height: 26,
+  padding: 0,
+  margin: theme.spacing(1),
+  '.MuiSwitch-switchBase': {
+    padding: 1,
+    '&.Mui-checked': {
+      transform: 'translateX(16px)',
+      color: theme.palette.common.white,
+      '& + .MuiSwitch-track': {
+        backgroundColor: '#52d869',
+        opacity: 1,
+        border: 'none',
       },
     },
-    thumb: {
-      width: 24,
-      height: 24,
-    },
-    track: {
-      borderRadius: 26 / 2,
-      border: `1px solid ${theme.palette.grey[400]}`,
-      backgroundColor: theme.palette.grey[50],
-      opacity: 1,
-      transition: theme.transitions.create(['background-color', 'border-color']),
+    '&.Mui-focusVisible .MuiSwitch-thumb': {
+      color: '#52d869',
+      border: '6px solid #fff',
     },
-    checked: {},
-    focusVisible: {},
-  }),
-)(({ classes, ...props }: Props) => {
-  return (
-    <Switch
-      focusVisibleClassName={classes.focusVisible}
-      disableRipple
-      classes={{
-        root: classes.root,
-        switchBase: classes.switchBase,
-        thumb: classes.thumb,
-        track: classes.track,
-        checked: classes.checked,
-      }}
-      {...props}
-    />
-  );
-});
+  },
+  '.MuiSwitch-thumb': {
+    width: 24,
+    height: 24,
+  },
+  '.MuiSwitch-track': {
+    borderRadius: 26 / 2,
+    border: `1px solid ${theme.palette.grey[400]}`,
+    backgroundColor: theme.palette.grey[50],
+    opacity: 1,
+    transition: theme.transitions.create(['background-color', 'border-color']),
+  },
+}));
 
-const AntSwitch = withStyles((theme: Theme) =>
-  createStyles({
-    root: {
-      width: 28,
-      height: 16,
-      padding: 0,
-      display: 'flex',
-    },
-    switchBase: {
-      padding: 2,
-      color: theme.palette.grey[500],
-      '&$checked': {
-        transform: 'translateX(12px)',
-        color: theme.palette.common.white,
-        '& + $track': {
-          opacity: 1,
-          backgroundColor: theme.palette.primary.main,
-          borderColor: theme.palette.primary.main,
-        },
+const AntSwitch = styled(Switch)(({ theme }) => ({
+  width: 28,
+  height: 16,
+  padding: 0,
+  display: 'flex',
+  '.MuiSwitch-switchBase': {
```

Comment:
How about we keep using & systematically?

I don't even know if it's equivalent to
```suggestion
  '&.MuiSwitch-switchBase': {
```

or 

```suggestion
  '& .MuiSwitch-switchBase': {
```


x the other cases

Reply:
Will check tomorrow 😁

Reply:
Resolved. In the first way the styles are broken, but in the second it works 🎉